pkg/tsdb/cloudwatch/client_factory.go

```diff
@@ -52,8 +52,8 @@ var NewCWLogsClient = func(sess *session.Session) cloudwatchlogsiface.CloudWatch
 // EC2 client factory.
 //
 // Stubbable by tests.
-var newEC2Client = func(provider client.ConfigProvider) models.EC2APIProvider {
-	return ec2.New(provider)
+var NewEC2Client = func(sess *session.Session) models.EC2APIProvider {
+	return ec2.New(sess)
 }
```

Comment:
this change feels the shadiest to me. I don't really understand how this is so easily changed without affecting things? maybe it will become more apparent in tests?

Reply:
yeah I think it doesn't need to change, I don't really know why I did this, putting it back.